framework/src/actions/SplitMeshAction.C

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SplitMeshAction.h"
+
+#include "MooseApp.h"
+#include "MooseUtils.h"
+#include "MooseMesh.h"
+#include "libmesh/checkpoint_io.h"
+
+template <>
+InputParameters
+validParams<SplitMeshAction>()
+{
+  return validParams<Action>();
+}
+
+SplitMeshAction::SplitMeshAction(InputParameters params) : Action(params) {}
+
+void
+SplitMeshAction::act()
+{
+  auto mesh = _app.actionWarehouse().mesh();
+
+  if (mesh->getFileName() == "" && _app.parameters().get<std::string>("split_file") == "")
+    mooseError("must specify an output mesh file name (--split-file) when splitting "
+               "non-file-based meshes");
+
+  auto splitstr = _app.parameters().get<std::string>("split_mesh");
+  std::vector<unsigned int> splits;
+  bool success = MooseUtils::tokenizeAndConvert(splitstr, splits, ", ");
+  if (!success)
+    mooseError("invalid argument for --split-file: '", splitstr, "'");
+
+  for (std::size_t i = 0; i < splits.size(); i++)
+  {
+    processor_id_type n = splits[i];
+    Moose::out << "splitting " << n << " ways..." << std::endl;
+
+    auto cpr = libMesh::split_mesh(*mesh, n);
+    Moose::out << "    * writing " << cpr->current_processor_ids().size() << " files per process..."
```

Comment:
Note: OK, but we really should revisit the whole console stream interface so that we can use it more consistently. I know you can't here because your messages would be buffered.